File: reorder_numerals.py
import json
import re

# 读取文件
with open('words-data.json', 'r', encoding='utf-8') as f:
    content = f.read()

# 解析JSON
data = json.loads(content)
words = data['words']

# 分离数词和其他单词
numerals = []
others = []

for word in words:
    if word.get('pos') == 'numeral':
        numerals.append(word)
    else:
        others.append(word)

print(f"找到 {len(numerals)} 个数词")
print(f"其他单词: {len(others)} 个")

# 定义数词顺序
numeral_order = [
    # 基数词 0-19
    'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine',
    'ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen',
    'seventeen', 'eighteen', 'nineteen',
    # 整十基数词
    'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety',
    # 大数
    'hundred', 'thousand', 'million',
    # 序数词
    'first', 'second', 'third', 'fourth', 'fifth', 'sixth', 'seventh', 'eighth', 'ninth', 'tenth',
    # 其他数量词
    'half', 'quarter', 'double', 'triple', 'dozen', 'single', 'pair', 'couple', 'several', 'both'
]

# 创建数词字典
numeral_dict = {n['word']: n for n in numerals}

# 按顺序排列数词
sorted_numerals = []
for word_name in numeral_order:
    if word_name in numeral_dict:
        sorted_numerals.append(numeral_dict[word_name])
        print(f"添加: {word_name}")
    else:
        print(f"警告: 未找到 {word_name}")

print(f"\n排序后数词数量: {len(sorted_numerals)}")

# 找到插入位置（在interjection之后，preposition之前）
insert_index = 0
for i, word in enumerate(others):
    if word.get('pos') == 'interjection':
        insert_index = i + 1

print(f"插入位置: {insert_index}")

# 合并列表
new_words = others[:insert_index] + sorted_numerals + others[insert_index:]

print(f"新单词总数: {len(new_words)}")

# 保存回文件
data['words'] = new_words
with open('words-data.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=4)

print("\n完成！数词已按数字顺序重新排列。")

File: reorder_numerals.js
const fs = require('fs');

// 读取JSON文件
const data = JSON.parse(fs.readFileSync('words-data.json', 'utf8'));
const words = data.words;

// 分离数词和其他单词
const numerals = [];
const others = [];

for (const word of words) {
    if (word.pos === 'numeral') {
        numerals.push(word);
    } else {
        others.push(word);
    }
}

console.log(`Found ${numerals.length} numerals`);
console.log(`Other words: ${others.length}`);

// 定义数词顺序
const numeralOrder = [
    // 基数词 0-19
    'zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine',
    'ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen',
    'seventeen', 'eighteen', 'nineteen',
    // 整十基数词
    'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety',
    // 大数
    'hundred', 'thousand', 'million',
    // 序数词
    'first', 'second', 'third', 'fourth', 'fifth', 'sixth', 'seventh', 'eighth', 'ninth', 'tenth',
    // 其他数量词
    'half', 'quarter', 'double', 'triple', 'dozen', 'single', 'pair', 'couple', 'several', 'both'
];

// 创建数词字典
const numeralDict = {};
for (const n of numerals) {
    numeralDict[n.word] = n;
}

// 按顺序排列数词
const sortedNumerals = [];
for (const wordName of numeralOrder) {
    if (numeralDict[wordName]) {
        sortedNumerals.push(numeralDict[wordName]);
        console.log(`Added: ${wordName}`);
    } else {
        console.log(`Warning: Not found ${wordName}`);
    }
}

console.log(`\nSorted numerals count: ${sortedNumerals.length}`);

// 找到插入位置（在interjection之后）
let insertIndex = 0;
for (let i = 0; i < others.length; i++) {
    if (others[i].pos === 'interjection') {
        insertIndex = i + 1;
    }
}

console.log(`Insert position: ${insertIndex}`);

// 合并列表
const newWords = [
    ...others.slice(0, insertIndex),
    ...sortedNumerals,
    ...others.slice(insertIndex)
];

console.log(`Total words: ${newWords.length}`);

// 保存回文件
data.words = newWords;
fs.writeFileSync('words-data.json', JSON.stringify(data, null, 4), 'utf8');

console.log('\nDone! Numerals reordered by number sequence.');
